AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -138,60 +100,69 @@ class CompleteScreenViewModel: ObservableObject {
         exitRoute()
     }
     
+    func confirmationButtonPressed() {
+        guard let externalSessionWithStreams = externalSessionWithStreams else {
+            assertionFailure("Confirmation button pressed when there was no session with streams")
+            return
+        }
+        
+        Log.info("session: \(externalSessionWithStreams)")
+        
+        do {
+            try externalSessionsStore.createExternalSession(session: externalSessionWithStreams)
+            // TODO: remove after debugging
+            let s = try externalSessionsStore.getExistingSession(uuid: externalSessionWithStreams.uuid)
+            Log.info("\(s.measurementStreams)")
+            completeButtonEnabled = false
+            completeButtonText = followedText
+        } catch {
+            Log.error("FAILED: \(error)")
+            self.alert = InAppAlerts.failedSessionDownloadAlert(dismiss: self.dismissView)
+        }
+        
+    }
+    
     private func dismissView() {
         exitRoute()
     }
     
-    private func getMeasurementsAndDisplayData(_ thresholds: ThresholdsValue) {
-        let streams = [String(session.streamID)] // In the future this will be changed for Airbeam sessions, cause we will need to get other streams ids from backend
-        downloadMeasurements(streams: streams) { [weak self] result in
+    private func getMeasurementsAndDisplayData() {
+        let streams = session.stream.map(\.id)
+        
+        controller.downloadMeasurements(streams: streams) { [weak self] result in
```

Comment:
continuing with `Controller` purpose - a lot of this code here looks like application logic and it seems like it can be packed into a `Controller`. See how it first uses `downloadMeasurements` here and than after result is present it again calls the `Controller` with `createExternalSession`. Maybe it would be possible to refactor this VM so that most of this work gets done inside some single high level function inside the `Controller`? It will probably require some rethinking and restructuring here, but it would clear the field here a lot and make this ViewModel uncluttered (cuz right now a lot of code here is not really view-related and can be a struggle to get thru if someone just wanted to quickly get an idea how the UI works)